# Cutting a Stick Riddle

Given a stick of length $L$, cut it at will in two parts. 
Measure the ratio between the shorter and the longer parts.
Repeat the procedure, cutting it at random each time.
What is the expected value of the ratio?

## Brute force solution

Well, since this is a Notebook and we have the CPU power to aid us, let us first solve it very easily by literally repeating the experiment $N$ times, computing the ratio $\rho=L_\mathrm{short}/L_\mathrm{long}$ and taking the average $s=\sum \rho_i / N$.

It is useful to notice that $L_\mathrm{short} = L-L_\mathrm{long}$.

For fun let's solve it in a single line of code with $N=10000$:

In [2]:
from numpy import random
"%.4f" % (np.sum([min(1-rnd, rnd)/max(1-rnd,rnd) for rnd in random.rand(10000)])/10000)

'0.3862'

## Analytical treatment

Now, for a little more sophisticated solution.

Let $x=L_\mathrm{long}$ be the random variable of the problem. 
The ratio is $\rho(x) = \frac{1-x}{x}$.
Since the long side is the independent varible, the function is defined in the interval $\rho: x \in [1/2,1] \to [0,1]$.
The problem states that we cut at random, so each point in the interval $[0,L]$ has the same probability of being picked, i.e. the probability distribution from wich $x$ is drawn is constant $p(x)=1/\mu(x)=2$, where $\mu(x)=1/2$  is the measure of the domain of $x$.

We now have all the ingredients to get the expectation value:
\begin{align}
    \langle \rho \rangle &= \int_{1/2}^{1} \rho(x) p(x) dx \\
        &= 2 \int_{1/2}^{1} \frac{1-x}{x} dx \\
        &= 2 \log2 -1 \approx 0.386
\end{align}

We can check that computed and numberical solutions agree

In [9]:
from time import time
avg_anal = 2*log(2)-1
N = int(1e8) # Sample size for the estimator. Don't go too big, already a 100mil take two min.
t0 = time()
avg_estimator = np.sum([min(1-rnd, rnd)/max(1-rnd,rnd) for rnd in random.rand(N)])/N
print("Analytical solution %.4f." % avg_anal)
print("Numerical estimation %.4f." % avg_estimator)
print("Relative difference %.2g %%" % (100*(avg_anal-avg_estimator)/avg_anal))
t1 = time()
print("Done in %is (%.2fmin)" %(t1-t0,(t1-t0)/60))

Analytical solution 0.3863.
Numerical estimation 0.3863.
Relative difference -0.0062 %
Done in 120s (2.00min)
